Build errors with esp-idf instead of arduino framework

When building with the arduino framework we see:
```
In file included from src/main.cpp:122:
src/leapmmw_sensor.h: In member function 'virtual void leapmmw::loop()':
src/leapmmw_sensor.h:167:28: warning: 'setled_state' may be used uninitialized in this function [-Wmaybe-uninitialized]
  167 |               publishSwitch("led", setled_state);
      |               ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
```

When building with the esp-idf framework this is treated as an error:
```
In file included from src/main.cpp:107:
src/leapmmw_sensor.h: In member function 'virtual void leapmmw::loop()':
src/leapmmw_sensor.h:167:28: error: 'setled_state' may be used uninitialized in this function [-Werror=maybe-uninitialized]
               publishSwitch("led", setled_state);
               ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
[...snip...]
cc1plus: some warnings being treated as errors
*** [/data/devtest/.pioenvs/devtest/src/main.o] Error 1
```

## Workaround
While this may break LED functionality comment out line 167 in the .h file.
### From
`publishSwitch("led", setled_state);`
### To
`// publishSwitch("led", setled_state);`

I haven't tested the workaround with real hardware because I don't have an esp-idf board with the sensor attached. Will work on getting that set up.

--- a/leapmmw_sensor.h
+++ b/leapmmw_sensor.h
@@ -161,7 +161,7 @@ class leapmmw : public Component, public UARTDevice {
             } else {
               int led_state = parse_number<int>(getLedMode).value();
               // ESP_LOGD("custom", "The value of getLedMode is: %i", led_state);
-              int setled_state;
+              int setled_state = -1;
               if (led_state == 1) {
                 setled_state = 0;
               } else if (led_state == 0) {

